shared/macros-kubernetes.jinja

```diff
@@ -162,8 +168,16 @@ spec:
     storage:
       files:
       - contents:
-          source: data:,{{{ url_encode(SOURCE) }}}
+          source: data:,{{{ url_encode(source) }}}
         filesystem: root
-        mode: {{{ MODE }}}
-        path: {{{ PATH }}}
+        mode: {{{ file_permissions_mode }}}
+        path: {{{ path }}}
+{{%- endmacro -%}}
+
+
+{{#
+  High level macro to generate Kubernetes remediation to set the ssh daemon configuration file.
+#}}
+{{%- macro kubernetes_sshd_set() -%}}
+{{{ kubernetes_machine_config(path='/etc/ssh/sshd_config', file_permissions_mode='0600', source=sshd_config_source()) }}}
```

Comment:
can we switch the macro from kubernetes_machine_config to kubernetes_machine_config_file or something of the sort? we could also start doing similar macros for systemd units, for instance. which requires a different template.

Reply:
yes, I can do that.

Reply:
Done